### PR TITLE
Add self-signed cert verification (#4)

### PR DESCRIPTION
**Pull request by sgnn7**
This PR:
- Adds ability to consume provided server certificates and related tests
- Makes test suite use https endpoint for tests
- Refactors test suite to be less fragile
- Enhances test suite to be able to consume env vars for controller
- Squelches unnecessary pytest output
- Bumps version to v1.0.5
- Removes cruft files from repo
- Adds build_release script
- Adds jenkins archiving of release artifacts on main branch

Connected to #4
Connected to #14

### CODE REVIEW

File: plugins/lookup/conjur_variable.py [hunk 1]
```diff
@@ -134,24 +137,30 @@ def _load_identity_from_file(identity_path, appliance_url):
...
 # slashes correctly) because it encodes spaces into the character '+' instead of "%20"
 # as expected by the Conjur server
-def _encode_str(str):
-    return quote(str, safe='')
+def _encode_str(input_str):
+    return quote(input_str, safe='')
 
 # Use credentials to retrieve temporary authorization token
-def _fetch_conjur_token(conjur_url, account, username, api_key, validate_certs):
+def _fetch_conjur_token(conjur_url, account, username, api_key, validate_certs, cert_file):
```
Comment (izgeri) [inline plugins/lookup/conjur_variable.py:153]: not sure how python works - does this method still work if less than this number of arguments is supplied, or is this a backwards-incompatible change?

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,12 +5,16 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Added
+- Plugin now can correctly consume provided server certificates
```
Comment (izgeri) [inline CHANGELOG.md:9]: is it CA certs that it's consuming?

is it more correct to say "plugin supports validation of self-signed certificates using provided `CONJUR_CERT_FILE` env var"? (don't know if I'm understanding how it's configured correctly)

Comment (izgeri) [reply CHANGELOG.md:9]: also I wish the cert env var was consistent with the rest of our products, but I'll file a follow-up issue for that

**Comment by izgeri** [comment]
do you want to add a commit that bumps the version so that you can tag once this PR is merged?

File: plugins/lookup/conjur_variable.py [hunk 3]
```diff
@@ -1,4 +1,4 @@
-# (c) 2018, Jason Vanderhoof <jason.vanderhoof@cyberark.com>, Oren Ben Meir <oren.benmeir@cyberark.com>
+# (c) 2020, Jason Vanderhoof <jason.vanderhoof@cyberark.com>, Oren Ben Meir <oren.benmeir@cyberark.com>, Srdjan Grubor <sgnn7@sgnn7.org>
```
[the remark this answers is not in the document]
Comment (sgnn7) [reply plugins/lookup/conjur_variable.py:1]: probably but I can just ad conj_manintainers

File: plugins/lookup/conjur_variable.py (hunk 1 above)
Comment (sgnn7) [reply plugins/lookup/conjur_variable.py:153]: Technically it's a backwards-incompatible change but this is a private method (`_` prefix) so there should not be anything other than this module using this method.

File: CHANGELOG.md (hunk 2 above)
Comment (sgnn7) [reply CHANGELOG.md:9]: I'll update though it's not specific to that exact var (the ssl cert can be specified in conjur.conf too)

**Comment by sgnn7** [comment]
Fixed up and added commits for v1.0.5 prep

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -5,3 +5,16 @@ Thanks for your interest in the Ansible Conjur collection.
...
+
+## Releasing
+
+To release a new version of this plugin:
+
+- Make the appropriate changes
+- Update the version number in [`galaxy.yml`](galaxy.yml) and [`CHANGELOG.md`](CHANGELOG.md)
+- Tag the git history with `v##.##.##` version
+- Create the release on GitHub for that tag
+- Build the release package with `./bin/build_release`
```
[the remark this answers is not in the document]
Comment (sgnn7) [reply CONTRIBUTING.md:17]: It can be but that part isn't tested yet so I'd avoid mentioning it explicitly here yet

**Comment by sgnn7** [comment]
@izgeri 

> there is a cyberark-conjur-1.0.4.tar.gz binary on this branch now - I'm guessing that's an accident?

This file was on main branch. This PR removes it :)

**Pull request merged**